### PR TITLE
Fallow Mateusz code review suggestions.

### LINKED ISSUE #32: Changes to Island class based on Mateusz code review.

### PR DESCRIPTION
**Pull request by runer53248**
Insert coordinate type into Island class. 
Island::Coordinate can be constexpr now. 
Made Island::getPosition constexpr as well. 
Change Coordinates::toString to friend operator<< for Coordinates.
Remove if-else in Island::Coordinates::operator==.

### CODE REVIEW

File: shm/Island.hpp [hunk 1]
```diff
@@ -1,27 +1,28 @@
 #pragma once
 #include <cstdint>
 #include <string>
-
-using coordinateType = std::size_t;
-
+#include <iostream>
 class Island {
 public:
+    using coordinateType = std::size_t;
```
Comment (Martyna-Tu) [inline shm/Island.hpp:7]: Why don't you put it in Coordinates class ??

File: shm/Island.hpp [hunk 2]
```diff
@@ -1,27 +1,28 @@
 #pragma once
 #include <cstdint>
 #include <string>
-
-using coordinateType = std::size_t;
-
+#include <iostream>
 class Island {
```
Comment (Martyna-Tu) [inline shm/Island.hpp:5]: Separate includes form class

File: shm/Map.cpp [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #include "Map.hpp"
 #include <algorithm>
 
-coordinateType generatePosition() {
+Island::coordinateType generatePosition() {
```
Comment (Martyna-Tu) [inline shm/Map.cpp:4]: line 29 in Map.ccp - move islandVec_.reserve(...) from c-tor to fillWithRandomIslands() - it was Mateusz suggestion

Comment (Rogal255) [reply shm/Map.cpp:4]: I think this PR is to fix Island and Coordinates clases only. Here the change is only to access coordinateType which is now in the Island class as Mateusz suggested. For me this conversation can be resolved as I consider that everything is OK here.

**Review by Rogal255 — APPROVED** [review]
The code itself is OK however fix formatting please.

**Pull request merged**